mhs/mhs/common/state/tests/test_dynamo_persistence_adaptor.py

```diff
@@ -30,146 +28,213 @@
-            await self.service.delete(TEST_INVALID_TABLE, TEST_KEY)
-
-        self.assertIsInstance(ex.exception.__cause__, InvalidTableError)
-
-    @async_test
-    async def test_delete_invalid_key(self):
+    async def test_delete_invalid_key(self, mock_boto3_resource):
+        """Test unhappy path for delete where a invalid/missing key is provided."""
+        self.__configure_mocks(mock_boto3_resource)
         self.__setFutureResponse(self.mock_table.delete_item, TEST_DELETE_EMPTY_RESPONSE)
 
-        response = await self.service.delete(TEST_STATE_TABLE, TEST_INVALID_KEY)
+        response = await self.service.delete(TEST_INVALID_KEY)
 
         self.assertIsNone(response)
 
+    @patch.object(dynamo_persistence_adaptor.aioboto3, "resource")
     @async_test
-    async def test_delete_io_exception(self):
+    async def test_delete_io_exception(self, mock_boto3_resource):
+        """Test unhappy path for delete where an IO exception occurs."""
+        self.__configure_mocks(mock_boto3_resource)
         self.mock_table.delete_item.side_effect = TEST_SIDE_EFFECT
 
         with self.assertRaises(RecordDeletionError) as ex:
-            await self.service.delete(TEST_STATE_TABLE, TEST_INVALID_KEY)
+            await self.service.delete(TEST_INVALID_KEY)
 
         self.assertIs(ex.exception.__cause__, TEST_EXCEPTION)
 
     @staticmethod
-    def __setFutureResponse(method, response):
+    def __setFutureResponse(method, response) -> None:
+        """
+        Utility method to help setting expected responses on an async/await call.
+        :param method: The method to add the response to.
+        :param response: The response to be added to the method.
+        """
         future = asyncio.Future()
         future.set_result(response)
 
         method.return_value = future
+
+    def __configure_mocks(self, mock_boto3_resource):
+        """Configure the standard mocks to have mappings which can be used in the tests."""
+        mock = MagicMock()
+        self.mock_table = mock.Table.return_value
+        mock_boto3_resource.return_value = constant_context_manager(mock)
+
+
+@contextlib.asynccontextmanager
+async def constant_context_manager(mock: Mock):
+    """
+    Create a context manager which will provide a mock.
+    :param mock: The mock to provide.
+    :return: The mock required.
+    """
+    yield mock
+
+
+def single_loop_async_test(f):
+    """
+    A wrapper for asynchronous tests.
+    By default unittest will not wait for asynchronous tests to complete even if the async functions are awaited.
+    By annotating a test method with `@async_test` it will cause the test to wait for asynchronous activities
+    to complete. This will only use a single event loop for all tests with this annotation.
+    :param f: The coroutine which will be executed.
+    :return: a function which will execute an asynchronous unit test.
+    """
+
+    def wrapper(*args, **kwargs):
```

Comment:
Optional: This is being pedantic, and you could argue isn't really required for a test, but can you stick `@wraps(f)` as a decorator for this function. [Python docs here](https://docs.python.org/3/library/functools.html#functools.wraps). It's the default way that decorators are created in the Python world.

Reply:
done.